.github/workflows/workflow.yml

```diff
@@ -0,0 +1,65 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow will install Deno and run tests across stable and canary builds on Windows, Ubuntu and macOS.
+# For more information see: https://github.com/denoland/setup-deno
+
+name: CodersCampWorkflow
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build-and-run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: install dependencies 
+        run: yarn install --immutable
+      - name: build api
+        run: yarn w api build
+      - name: build client
+        run: yarn w website build
+      - name: run affected tests 
+        run: yarn test --onlyChanged
+      
+  es-Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: install dependencies 
+        run: yarn install --immutable
+      - name: Run ESLint
+        run: yarn eslint --cache
+  Typescript-compiler:
+    name: tsc
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: install node v14
+      uses: actions/setup-node@v1
+      with:
+        node-version: 14
+    - name: install dependencies
+      run: yarn install
+    - name: tsc compile webside
+      uses: iCrawl/action-tsc@v1
```

Comment:
It looks like this action is no longer maintained (its repository is archived).
Maybe it would be better to define the `typecheck` command for each package and then add `yarn workspaces foreach -vpi run typecheck` script to the main `package.json` file. What do you think about it?